services/src/main/java/org/fao/geonet/inspire/validator/InspireValidatorUtils.java

```diff
@@ -254,20 +256,22 @@ private String uploadMetadataFile(ServiceContext context, String endPoint, Input
 
             response = this.execute(context, request);
 
-            if (response.getStatusCode().value() == 200) {
+            if (response.getStatusCode().value() == HttpStatus.SC_OK) {
 
                 new BasicResponseHandler();
                 String body = CharStreams.toString(new InputStreamReader(response.getBody()));
                 JSONObject jsonRoot = new JSONObject(body);
                 return jsonRoot.getJSONObject("testObject").getString("id");
             } else {
-                Log.warning(Log.SERVICE,
-                    "WARNING: INSPIRE service HTTP response: " + response.getStatusCode().value() + " for " + TestObjects_URL);
-                return null;
+                Log.error(Log.SERVICE,
+                    "INSPIRE service HTTP response: " + response.getStatusCode().value() + " for " + TESTOBJECTS_URL);
+                throw new InspireValidatorException(String.format("INSPIRE service - error uploading file: %s, error code: %d", response.getStatusText(), response.getStatusCode().value()));
             }
+        } catch (InspireValidatorException e) {
+            throw e;
         } catch (Exception e) {
             Log.error(Log.SERVICE, "Error calling INSPIRE service: " + endPoint, e);
-            return null;
+            throw new InspireValidatorException(String.format("Error calling INSPIRE service: %s, %s", endPoint, e.getMessage()));
```

Comment:
I'd pass the full exception `e` to `new InspireValidatorException()` constructor to allow the logger to print `Caused by: ...` and get the full stacktrace.